### PR TITLE
Mlucas docu 1371

### PR DESCRIPTION
**Pull request by lucasdigital**
Adding IBM COS as a source. See DOCU-1371 for the slow accumulation of info.
My main concern is the command-line examples. The latter ones may be misinterpreted or misrepresented.

### CODE REVIEW

File: docs/command-reference.md [hunk 1]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
+* Adding file to container
+
+  ```text
+  ./mc cp ~/Downloads/wq4.pptx cos/container2/
+  ```
+
+* Removing a file from a container
+
+  ```text
+  ~/Downloads/minio$ ./mc rm cos/container2/wq4.pptx
```
Comment (fi3ld) [inline docs/command-reference.md:553]: Remove ~/Downloads/minio$

File: docs/command-reference.md [hunk 2]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
+    "sourceFileSystem": "cos_s3_source2",
+    "sourcePath": "/",
+    "targetFileSystem": "targetHdfs2",
+    "targetPath": "/repl_test1"
+  ```
+
+* Adding file to container
+
+  ```text
+  ./mc cp ~/Downloads/wq4.pptx cos/container2/
```
Comment (fi3ld) [inline docs/command-reference.md:547]: Remove ./mc

File: docs/command-reference.md [hunk 3]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
 
-#### Example
+#### S3a Example
+
+```text
+filesystem add s3a --file-system-id mytarget
+--bucket-name mybucket1
+--credentials-provider org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider
+--access-key B6ZAI18Z3UIO002Y777A
+--secret-key OP87Chokisf4hsTP0Q5j95yI904lT7AaDBGJpp0D
```
Comment (fi3ld) [inline docs/command-reference.md:474]: Are these credentials still valid? The access-key/secret-key must be invalidated and deleted before publishing.

File: docs/command-reference.md [hunk 4]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
+#### IBM COS Examples
+
+ * Add source IBM COS filesystem. Note that this does not work if SSL is used on the endpoint address.
+
+  ```text
+  filesystem add s3a --source --file-system-id cos_s3_source2
+  --bucket-name container2
+  --credentials-provider org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider
+  --access-key pkPgjLaOMiGCAtlZrViz
+  --secret-key c3vq6vaNtTIpd8uEDnDHRpHeeVuqJMIHuV9IF3n9
```
Comment (fi3ld) [inline docs/command-reference.md:486]: Invalidate and delete secret key

File: docs/command-reference.md (hunk 1 above)
Comment (fi3ld) [reply docs/command-reference.md:553]: also remove .mc

File: docs/command-reference.md [hunk 5]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
+* Removing a file from a container
+
+  ```text
+  ~/Downloads/minio$ ./mc rm cos/container2/wq4.pptx
+  ```
+
+* List objects in container
+
+  ```text
+  ./mc ls cos/container2/
```
Comment (fi3ld) [inline docs/command-reference.md:559]: Remove .mc

File: docs/command-reference.md [hunk 6]
```diff
@@ -464,11 +464,121 @@ Both the `array` and `bytebuffer` options may lead to the consumption of large a
...
+
+  ```text
+  aws s3api list-objects --endpoint-url=http://10.0.0.201
+  --bucket container2
+  ```
+
+* config mc
+
+  ```title
+  nano ~/.mc/config.json
```
Comment (fi3ld) [inline docs/command-reference.md:572]: Is this needed? If so, where will the user likely find config.json? 

We don't need nano

**Comment by lucasdigital** [comment]
Fixes in place. References to Minio removed and AWS secret credentials obfiscated.

**Pull request merged**